yt_dlp/extractor/stacommu.py

```diff
@@ -0,0 +1,154 @@
+import time
+
+from .wrestleuniverse import WrestleUniverseBaseIE
+from ..utils import (
+    int_or_none,
+    traverse_obj,
+    url_or_none,
+)
+
+
+class StacommuBaseIE(WrestleUniverseBaseIE):
+    _NETRC_MACHINE = 'stacommu'
+    _API_HOST = 'api.stacommu.jp'
+    _API_PATH = None
+    _REAL_TOKEN = None
+    _TOKEN_EXPIRY = None
+    _REFRESH_TOKEN = None
+    _DEVICE_ID = None
+    _LOGIN_QUERY = {'key': 'AIzaSyCR9czxhH2eWuijEhTNWBZ5MCcOYEUTAhg'}
+    _LOGIN_HEADERS = {
+        'Accept': '*/*',
+        'Content-Type': 'application/json',
+        'X-Client-Version': 'Chrome/JsCore/9.9.4/FirebaseCore-web',
+        'Referer': 'https://www.stacommu.jp/',
+        'Origin': 'https://www.stacommu.jp',
+    }
+
+    @WrestleUniverseBaseIE._TOKEN.getter
+    def _TOKEN(self):
+        if self._REAL_TOKEN and self._TOKEN_EXPIRY <= int(time.time()):
+            self._refresh_token()
+
+        return self._REAL_TOKEN
```

Comment:
Not sure about this. Why not support cookies? If stacommu and wrestleuniverse have nearly identical backends, I'd be surprised if the auth token was not stored as a cookie. And if it is a cookie, then we could just move the hardcoded wrestleuniverse domain and cookiename out of the property/getter and into class vars, and we wouldn't need to redefine `_TOKEN` at all

Reply:
Different to wrestleuniverse, the token is not stored in cookie, but in IndexedDB.

Another reason to redefine _TOKEN is that the token is not necessary for free videos, which actually exist in stacommu. It seems all videos in wresleuniverse requires login, so its current implementation raises error without valid token.

Reply:
sending an authorization header of `Bearer None` isn't a problem for free videos?

Reply:
It works, but I think it should be fixed.

Reply:
I was only asking; I think https://github.com/yt-dlp/yt-dlp/pull/7432/commits/1bf19237e40e4012262b8b410761405a49cf13ac may have been too hasty. I don't want to change the behavior of `_call_api` for the wrestleuniverse extractors since I no longer have access to an account to test them with. I think a different approach is needed.

Also, the stacommu extractors need to `raise_login_required` with `method='password'` for premium content or else we are not letting users know that passing cookies does not work

Let me see what I can come up with